packages/x-data-grid-premium/src/components/pivotPanel/GridPivotPanelBody.tsx

```diff
@@ -0,0 +1,417 @@
+  }),
+  '&[data-drag-over="true"]': {
+    backgroundColor: vars.colors.interactive.hover,
+    outline: `2px solid ${vars.colors.interactive.selected}`,
+  },
+});
+
+const GridPivotPanelSectionTitle = styled('div', {
+  name: 'MuiDataGrid',
+  slot: 'PivotPanelSectionTitle',
+})<{ ownerState: OwnerState }>({
+  flex: 1,
+  marginRight: vars.spacing(1.75),
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  gap: vars.spacing(1),
+  font: vars.typography.font.body,
+  fontWeight: vars.typography.fontWeight.medium,
+});
+
+const GridPivotPanelFieldList = styled('div', {
+  name: 'MuiDataGrid',
+  slot: 'PivotPanelFieldList',
+})<{ ownerState: OwnerState }>({
+  flex: 1,
+  display: 'flex',
+  flexDirection: 'column',
+  padding: vars.spacing(0.5, 0),
+});
+
+const GridPivotPanelPlaceholder = styled('div', {
+  name: 'MuiDataGrid',
+  slot: 'PivotPanelPlaceholder',
+})<{ ownerState: OwnerState }>({
+  flex: 1,
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  textWrap: 'balance',
+  textAlign: 'center',
+  minHeight: 38,
+  height: '100%',
+  padding: vars.spacing(0, 1),
+  color: vars.colors.foreground.muted,
+  font: vars.typography.font.body,
+});
+
+export interface FieldTransferObject {
+  field: string;
+  modelKey: 'columns' | 'rows' | 'values' | null;
+}
+
+const INITIAL_DRAG_STATE = { active: false, dropZone: null, initialModelKey: null };
+
+function GridPivotPanelBody({ searchValue }: { searchValue: string }) {
+  const apiRef = useGridPrivateApiContext();
+  const initialColumns = useGridSelector(apiRef, gridPivotInitialColumnsSelector);
+  const fields = React.useMemo(() => Array.from(initialColumns.keys()), [initialColumns]);
+  const rootProps = useGridRootProps();
+  const [drag, setDrag] = React.useState<{
+    active: boolean;
+    dropZone: FieldTransferObject['modelKey'];
+    initialModelKey: FieldTransferObject['modelKey'];
+  }>(INITIAL_DRAG_STATE);
+  const pivotModel = useGridSelector(apiRef, gridPivotModelSelector);
+  const classes = useUtilityClasses(rootProps);
+
+  const getColumnName = React.useCallback(
+    (field: string) => {
+      const column = initialColumns.get(field);
+      return column?.headerName || field;
+    },
+    [initialColumns],
+  );
+
+  const availableFields = React.useMemo(() => {
+    return fields.filter((field) => {
+      if (pivotModel.rows.find((item) => item.field === field)) {
+        return false;
+      }
+      if (pivotModel.columns.find((item) => item.field === field)) {
+        return false;
+      }
+      if (pivotModel.values.find((item) => item.field === field)) {
+        return false;
+      }
+      if (searchValue) {
+        const fieldName = getColumnName(field);
+        return fieldName.toLowerCase().includes(searchValue.toLowerCase());
+      }
+      return true;
+    });
+  }, [pivotModel.columns, pivotModel.rows, pivotModel.values, searchValue, fields, getColumnName]);
```

Comment:
Could make it faster by avoiding redundant filtering in `availableFields` computation.
```suggestion
  const pivotFields = React.useMemo(() => new Set([
    ...pivotModel.rows.map((item) => item.field),
    ...pivotModel.columns.map((item) => item.field),
    ...pivotModel.values.map((item) => item.field),
  ]), [pivotModel]);
  
  const availableFields = React.useMemo(() => {
    return fields.filter((field) => {
      if (pivotFields.has(field)) {
        return false;
      }
      if (searchValue) {
        return getColumnName(field).toLowerCase().includes(searchValue.toLowerCase());
      }
      return true;
    });
  }, [pivotFields, searchValue, fields, getColumnName]);
```